src/app/services/authorization.service.ts

```diff
@@ -1,47 +1,77 @@
 import { Injectable, ChangeDetectorRef } from '@angular/core';
-import { Subject, Observable, of } from 'rxjs';
+import { Subject, Observable, of, Subscription } from 'rxjs';
 import { Consts } from '../consts/consts';
 import { Router } from '@angular/router';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { map } from 'rxjs/operators';
+import { createOfflineCompileUrlResolver } from '@angular/compiler';
 
 export interface IAuthMessage {
   isAuthentificated: boolean;
-  userName: string;
+  userName?: string;
+}
+
+interface ILoginResult {
+  token: string;
+}
+
+interface IUserInfo {
+  name: {
+    first: string,
+    last: string
+  };
 }
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthorizationService {
 
-  constructor(private router: Router) { }
+  constructor(private router: Router, private http: HttpClient) { }
+
+  public checkAuth = new Subject<boolean>();
+  public getUsername = this.getUserInfo().pipe(map(v => v.name.first + ' ' + v.name.last));
 
-  public refreshAuth() {
-    this.checkAuth.next({
-      isAuthentificated: this.isAuthentificated(),
-      userName: this.getUserInfo()
-    });
+  public isAuthentificated(): boolean {
+    return this.getToken() != null;
   }
 
-  public checkAuth = new Subject<IAuthMessage>();
+  public getToken(): string {
+    return localStorage.getItem(Consts.LS_TOKEN);
+  }
 
-  public isAuthentificated(): boolean {
-    return localStorage.getItem(Consts.LS_TOKEN) != null;
+  public setToken(token: string) {
+    localStorage.setItem(Consts.LS_TOKEN, token);
   }
 
-  public getUserInfo(): string {
-    return localStorage.getItem(Consts.LS_USERNAME);
+  public getUserInfo(): Observable<IUserInfo> {
+    return this.http.post<IUserInfo>(`http://localhost:3004/auth/userinfo`, {});
```

Comment:
Consider keeping api endpoints in environment.ts. To be able to set it depending on env.